### PR TITLE
Function for processing reverse proxy config.

### PR DESCRIPTION
**Pull request by ChriJ**
## Change
Added similar function to process, but with specified variables to swap from.

## Rationale
Allow use of configurations when going through the reverse proxy.

## Impact
Additional way of using configurations.

## Risk
- [x] The change does not increase the risk to the system and does therefore not require any extra risk analysis.
- [ ] A separate risk analysis has been performed and is linked below.

## Rollback
- [x] Rollback is performed by reverting the merge and redeploy.
- [ ] Rollback of this change requires special actions as outlined below.

### CODE REVIEW

File: Proxy/Configuration/Response/Json/index.ts [hunk 1]
```diff
@@ -130,6 +130,21 @@ export namespace Json {
...
+	): Record<string, any> {
+		return configuration.set.reduce((r, replacement) => {
+			if (typeof replacement == "string")
+				replacement = { find: replacement }
+			return Selector.set(
+				r,
+				replacement.find,
+				replacement.value == undefined ? undefined : replace(replacement.value, variables)
+			)
+		}, body)
```
Comment (lucymeachem-p2p) [inline Proxy/Configuration/Response/Json/index.ts:146]: this is copied and pasted from the above function, the above one could just call this this

File: Proxy/Configuration/Response/Json/index.ts [hunk 2]
```diff
@@ -130,6 +130,21 @@ export namespace Json {
 			)
 		}, body)
 	}
+	export function reverseProcess(
+		configuration: Json,
+		variables: Record<string, any>,
+		body: Record<string, any>
+	): Record<string, any> {
```
Comment (lucymeachem-p2p) [inline Proxy/Configuration/Response/Json/index.ts:137]: can you add docs to this please? ideally also the rest of the stuff in this class would be great
https://tsdoc.org/ same format as javadoc

**Review by lucymeachem-p2p — CHANGES_REQUESTED** [review]

**Review by simonmika — CHANGES_REQUESTED** [review]
Why can't we just search for all tokens and replace them with the part they reference? That way not much configuration is needed and we achieve maximum flexibility.

**Review by lucymeachem-p2p — APPROVED** [review]
perfect!

**Pull request closed without merging**